src/couch_views/src/couch_views_indexer.erl

```diff
@@ -87,7 +88,49 @@ init() ->
         design_opts => Mrst#mrst.design_opts
     },
 
-    update(Db, Mrst, State).
+    try
+        update(Db, Mrst, State)
+    catch
+        exit:normal ->
+            ok;
+        Error:Reason  ->
+            NewRetry = Retries + 1,
+            RetryLimit = retry_limit(),
+
+            if NewRetry < RetryLimit -> ok; true ->
+                    NewData = add_error(Error, Reason, Data),
+                    couch_jobs:finish(undefined, Job, NewData),
+                    exit(normal)
+            end,
+
+            DataErr = Data#{<<"retries">> := NewRetry},
+            StateErr = State#{job_data := DataErr, last_seq := <<"0">>},
```

Comment:
What effect does resetting the `last_seq` have, would it reprocess the changes feed from 0?

Saw that we start the changes feeds from `view_seq` https://github.com/apache/couchdb/blob/0d84fed2d9cfcf66056eb6147434d0eab0b47299/src/couch_views/src/couch_views_indexer.erl#L193 wonder if we should reset it to that instead...

Or do we intend it to reprocess the changes feed from 0

Reply:
I didn't really like adding this line but couldn't think of another approach. It doesn't reset the build changes feed to zero. It only sets the job data last_seq to zero. This is used for the subscription information here https://github.com/apache/couchdb/blob/prototype/fdb-layer/src/couch_views/src/couch_views_jobs.erl#L70

What happens is if I don't change that to zero and a build has failed the last_seq is set to undefined which becomes `<<"undefined">>` which passes `ViewSeq >= UpdateSeq` so then the build is considered successfully built. Otherwise I can add a case check for `<<"undefined">>` in couch_views_jobs?

Reply:
Ah that makes sense. Good call. Maybe a comment about couch_jobs subscribe behavior would work there.